File: chapter-4/java/demo-web/src/main/java/my/app/demo/Office.java
package my.app.demo;

import com.redis.om.spring.annotations.Indexed;
import com.redis.om.spring.annotations.Searchable;
import lombok.Data;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

@Data
@RequiredArgsConstructor(staticName = "of")
public class Office {

    @NonNull
    @Searchable(nostem = true)
    private String address;

    @NonNull
    @Indexed
    private String addressNumber;

    @NonNull
    @Indexed
    private String city;

    @NonNull
    @Indexed
    private String state;

    @NonNull
    @Indexed
    private String postalCode;

    @NonNull
    @Indexed
    private String country;
}

File: chapter-4/dotnetcore/dotnetcsharp/Office.cs
using Redis.OM.Modeling;

namespace dotnetcsharp;

public class Office
{
    [Searchable] public string? Address { get; set; }
    [Indexed] public string? AddressNumber { get; set; }
    [Indexed] public string? City { get; set; }
    [Indexed] public string? State { get; set; }
    [Indexed] public string? PostalCode { get; set; }
    [Indexed] public string? Country { get; set; }
}